sympy/printing/latex.py

```diff
@@ -529,7 +529,7 @@ def convert_args(args):
                         term_tex = r"\left(%s\right)" % term_tex
 
                     if _between_two_numbers_p[0].search(last_term_tex) and \
-                            _between_two_numbers_p[1].match(term_tex):
+                            _between_two_numbers_p[1].match(str(term)):
```

Comment:
Thanks for the fix! Can you add more tests that hit this line?

Reply:
I have added extra tests covering negative expressions and exponents

Reply:
Why does this had to use `str(term)` than `term_tex`?

Reply:
term_tex for fractions has `\frac{}{}` and the regex code is running a `match()` for strings that start with `{[0-9` so the `\frac` never gets picked up because `match()` only matches from the beginning and you can't come up with a regex (very complicated) for `\frac` because of the backslash, hence change the input itself to term rather than latex turn because in the end the regex is just checking for number

Reply:
I still get the impression that this is an workaround and it can't handle the second case

```
>>> latex(Mul(2, Mul(Rational(3, 4), evaluate=False), 5, evaluate=False))
'2 \\cdot \\frac{3}{4} \\cdot 5'
>>> latex(Mul(2, Mul(Rational(3, 4), 5, evaluate=False), evaluate=False))
'2 \\frac{3}{4} \\cdot 5'
```

Reply:
What should be the expected output for the second case?

Reply:
My opinion is that the second should have the parenthesis between the second and the third argument
and also some consistent way about applying `cdot`
I have no clear idea about how 'between number' was defined in the past and how consistent the rule was, but I think that it's time to clarify them before jumping on a implementation that may complicate the printing code.